### PR TITLE
Add the new global theme and some local files for new theme

### PR DESCRIPTION
**Pull request by budparr**

### CODE REVIEW

File: layouts/index.html [hunk 1]
```diff
@@ -1,12 +1,5 @@
...
-    {{ range .Site.RegularPages.ByLastmod.Reverse }}
-    {{ .Render "thumbnail"}}
-    {{ end }}
-</div>
-
-{{ partial "contribute.html" . }}
-{{ end }}
+{{ define "main" }}
+  {{ $section_to_display := first 1000 .Site.RegularPages.ByLastmod.Reverse }}
+  {{ $.Scratch.Add "section_to_display" $section_to_display }}
```
Comment (bep) [inline layouts/index.html]: Even though this works, and I don't see how it should not continue to do so, concurrency-wise, but I consider this an anti-pattern. The Scratch is built for a particular flaw in the Golang templates, and this isn't it. Pass the values you need as arguments (use dict) to the partial.

File: layouts/partials/pagelayout.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            Hugo Themes <span class="f4 mt2 mt0-l db dib-l">({{ with $.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .Title }})</span>
+
+          {{- end }}
+        </h1>
+        <div class="{{ .Site.Params.copyClass }} mid-gray f5 f4-l">
+          {{ .Content }}
+        </div>
+      </article>
+      <div class="flex flex-wrap justify-left pr4">
+        {{ $globalVariables := .Site.Data.globals }}
```
Comment (bep) [inline layouts/partials/pagelayout.html]: Everything starting with "$" is by definition a variable, so no need to say it again, "$globals" is better.

File: static/_headers [hunk 3]
```diff
@@ -1,4 +0,0 @@
-/*
```
Comment (bep) [inline static/_headers:1]: Could you undelete this file?

File: layouts/_default/single.html [hunk 4]
```diff
@@ -1,60 +1,52 @@
...
-
-                {{ with .Params.homepage}}
-                <a class="btn btn-dark btn-lg" href="{{.}}"><i class="fa fa-home"></i> Homepage</a>
-                {{ end }}
-
-
-            </div>
+{{ define "main" }}
+
+    {{ $currentPageUrl := .URL }}
```
Comment (bep) [inline layouts/_default/single.html]: Is this used? If so, the Permalink variants is probably what you want.

**Review by bep — CHANGES_REQUESTED** [review]
In general it looks very good. I have added a coiple of coments, the main being the Scratch/Partial one. Nor-amally I would said fine, but people look to these projects for copy-and-paste inspiration, so it would be good to get it right before it spreads.

File: layouts/_default/single.html (hunk 4 above)
Comment (bep) [reply layouts/_default/single.html]: I will take make some notes about the theme tomorrow (https://github.com/budparr/gohugo-theme)

File: static/_headers (hunk 3 above)
Comment (budparr) [reply static/_headers:1]: I had intended to move this into the theme so it could be used in all the sites. You prefer it local to the site?

Comment (bep) [reply static/_headers:1]: Ah, OK -- I didn't "see the theme" when I wrote this. Let it be in the theme.

File: layouts/index.html (hunk 1 above)
Comment (budparr) [reply layouts/index.html]: Good call, and just rushing on my part for the easiest way to solve the problem when I conceived of this pattern. I'll need to fix this in the gohugo theme, too.

File: layouts/partials/pagelayout.html (hunk 2 above)
Comment (budparr) [reply layouts/partials/pagelayout.html]: Good point, thanks.

File: static/_headers (hunk 3 above)
Comment (budparr) [reply static/_headers:1]: And, just so you know, it's not in the theme yet. I didn't think to add it until I saw yours here.

Comment (bep) [reply static/_headers:1]: OK.

**Comment by budparr** [comment]
I addressed the issues you mentioned (and thanks for the comments), and I'll add the relevant changes to the theme as well.

**Review by bep — APPROVED** [review]

**Comment by bep** [comment]
The "Tag:" logic seems to be slightly broken. I tried to push a fix to this branch, but no access.

**Comment by budparr** [comment]
What's the issue/fix? I'm happy to do it so I can learn from doing so.

**Comment by bep** [comment]
Other than that I see a couple minor issues we could fix before we publish it:

1) A way to turn off search until the docs goes live.
2) Same with News
3) Fix the links URLs in the header
4) "The Hugo Gopher is based on an original work by Renée French." I see no Gopher, so really no need for credit?
5) This may sound pity, but we've had this discussion before. The text "By spf13 and friends" should be changed to something more generic, ala "By the Hugo Authors". Steve is the founder, but that sentence does not capture the truth anymore.

**Comment by bep** [comment]
As to the tag fix:

```
-              <span class="f4 mt2 mt0-l db dib-l">({{ with $.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
+              <span class="f4 mt2 mt0-l db dib-l">({{ with $.context.Data.Singular }}{{ . | humanize -}}: ```

File: layouts/partials/pagelayout.html [hunk 5]
```diff
@@ -6,7 +6,7 @@
         <h1 class="primary-color-dark f2">
             Hugo Themes
             {{ if not .context.IsHome -}}
-              <span class="f4 mt2 mt0-l db dib-l">({{ with $.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
+              <span class="f4 mt2 mt0-l db dib-l">({{ with $.context.Data.Singular }}{{ . | humanize }}:{{ end }} {{ .context.Title }})</span>
```
Comment (bep) [inline layouts/partials/pagelayout.html]: You should also move the space before the end. I'm pretty sure I had that in my example. Now, if the with is false, there will be an extra space, so "( Blog)".

**Comment by budparr** [comment]
I'm not sure what you meant by this "Fix the links URLs in the header" though I may have done this as part of the other changes.

> "The Hugo Gopher is based on an original work by Renée French." I see no Gopher, so really no need for credit?

It does show up on the 404 page and the home page, so I think it's better to leave in.

**Comment by bep** [comment]
Yea, the Docs link pointed wrong, but now it is all good -- but now the images do not show up (i.e.  they exist, but does not show) ... Is this only me?

**Comment by bep** [comment]
I see some related JS errors:

```
app.bundle.js:8 Uncaught Error: Error: No input element in the page matches #search-input
    at Function.value (app.bundle.js:8)
    at new e (app.bundle.js:8)
    at t (app.bundle.js:15)
    at Object.<anonymous> (app.bundle.js:1)
    at t (app.bundle.js:1)
    at Object.<anonymous> (app.bundle.js:1)
    at t (app.bundle.js:1)
    at app.bundle.js:1
    at app.bundle.js:1
```

**Comment by budparr** [comment]
Looks like the JS errors are because I removed search. I don't think that's the source of the image issue.

**Comment by budparr** [comment]
Oh, actually, I take that back. It might be related to the script because I'm lazy-loading the images. I think I'll take the simple way out and just hide the search box instead of not rendering it.

**Comment by budparr** [comment]
@bep I'm working on prev/next links. I can hold off and do in a different PR, if you like, though it shouldn't be more than a few minutes.

**Comment by bep** [comment]
No need to do it in a different PR. Just shout my name when you feel all is done. No rush.

**Comment by bep** [comment]
One more thing: There is a "flickering effect" when you load a new page -- like the images are loaded twice. I don't see this in the current theme site, and it looks kind of funky. Looks like some HTTP 301 from "//images" to "/images".

**Comment by budparr** [comment]
@bep I fixed the URLs but I think the flickering effect was caused by an animation delay* I had in there, which I removed, if you could let me know if it's improved.

_* The animation is there because lazyloading shows the image title then the image, and I'd like it to be a smoother transition._

**Comment by bep** [comment]
It looks perfect now, thanks.

**Comment by bep** [comment]
But I think your sort change from Lastmod to Date is wrong.  We want the most recently updated themes first (we pick the dates from the Git commit log)

**Comment by budparr** [comment]
> But I think your sort change from Lastmod to Date is wrong. We want the most recently updated themes first (we pick the dates from the Git commit log)

I was going to ping you in gitter about that. Wrong of me to unilaterally make that change. My thinking was that the previous/next links should be consistent with how the main page is loaded, and to my knowledge there's no way to make them order like that. It's slightly discombulating to go from the main page then flip through the single pages that are in a different order. Only slightly though, so whatever is best; I'm happy to revert that change.

**Comment by bep** [comment]
> and to my knowledge there's no way to make them order like that.

There is, but it is badly undocumented. Steve added a feature some years ago, but the API is fairly confusing. But whatever you do, sort by Lastmod.

**Comment by bep** [comment]
Also: Having all the images in full width on the theme detail page is maybe not perfect. Look at http://localhost:1313/docuapi/

**Comment by budparr** [comment]
@bep I fixed the images. It was something I didn't understand about the css I was using.

I reverted the main page sorting. I took a glance around the Hugo repo and don't see anything about using prev/next in that context. Happy to take care of it if you can point me in the right direction.

Thanks for your assist on all these. Much appreciated.

**Comment by bep** [comment]
@budparr just leave the next/prev as is, it is on my list of things to fix in Hugo, so let us not show some undocumented API to the world and then remove it in a month ...

**Comment by bep** [comment]
I will not be able to do this tonight, but if you are happy with the current version (I will also take a quick lookover), I can do the merge and the adjustments to the scripts (we now have a submodule) and push the new version live tomorrow? 

In an ideal world, we should maybe get all sites out at once, but I think it could be motivational to get things moving a little. There have been too much talk and little walk on Gitter.

**Comment by budparr** [comment]
Sounds good to me, thank you.

**Pull request merged**